In [45]:
# First run src/data_sources/get_data.py to get the csv in your working directory.
import pandas as pd
import datetime
import json

In [2]:
# Set study window
start_date_string = "01/10/2000 00:00:00"
end_date_string = "30/04/2023 23:59:59"
start_date = datetime.datetime.strptime(
    start_date_string, "%d/%m/%Y %H:%M:%S")
end_date = datetime.datetime.strptime(
    end_date_string, "%d/%m/%Y %H:%M:%S")

In [3]:
# Import list of companies involved in all games published during time window. First need to run get_data.py
companies_df = pd.read_csv('game_companies.csv')
companies_df.head()

Unnamed: 0  id                     name    start_date  changed_company_id   
0           0   1          Electronic Arts  3.913920e+08                 5.0  \
1           1   2                  BioWare  7.915968e+08                 NaN   
2           2   4        Eidos Interactive  8.335008e+08              4940.0   
3           3   5  Interplay Entertainment  4.416768e+08                 NaN   
4           4   8                 2K Games  1.106611e+09                 NaN   

   parent   change_date  
0     NaN           NaN  
1     1.0           NaN  
2    26.0  1.257725e+09  
3     NaN           NaN  
4   139.0           NaN

In [4]:
# Check the data types of the columns.
companies_df.dtypes

Unnamed: 0              int64
id                      int64
name                   object
start_date            float64
changed_company_id    float64
parent                float64
change_date           float64
dtype: object

In [5]:
# Check how many unique companies were returned.
companies_df['id'].nunique()

12659

We see that there are 12659 unique companies that have either published or developed a video game since 1 October 2000.

In [6]:
# Check how many companies do not have data for start_date.
companies_df[companies_df.start_date.notnull()].shape[0]

1946

In [ ]:
# Get list of included companies.


There are 1946 companies for which the founding date of the company is known.

In [11]:
# Filter out companies with a parent as the parent will be publicly listed.
companies_no_child_df = companies_df[~companies_df.parent.notnull()]
companies_no_child_df.head()
# companies_no_child_df.shape[0]

Unnamed: 0  id                     name   start_date  changed_company_id   
0           0   1          Electronic Arts  391392000.0                 5.0  \
3           3   5  Interplay Entertainment  441676800.0                 NaN   
6           6  12       Black Isle Studios  828230400.0                 NaN   
7           7  17         Digital Extremes  757296000.0                 NaN   
9           9  23        Feral Interactive  851990400.0                 NaN   

   parent  change_date  
0     NaN          NaN  
3     NaN          NaN  
6     NaN          NaN  
7     NaN          NaN  
9     NaN          NaN

In [19]:
# Get dataframe of companies which are either parents or do not have parents.
df_parents_mixed = companies_df[companies_df.parent.isnull()]
df_parents_mixed.head()

Unnamed: 0  id                     name   start_date  changed_company_id   
0           0   1          Electronic Arts  391392000.0                 5.0  \
3           3   5  Interplay Entertainment  441676800.0                 NaN   
6           6  12       Black Isle Studios  828230400.0                 NaN   
7           7  17         Digital Extremes  757296000.0                 NaN   
9           9  23        Feral Interactive  851990400.0                 NaN   

   parent  change_date  
0     NaN          NaN  
3     NaN          NaN  
6     NaN          NaN  
7     NaN          NaN  
9     NaN          NaN

493 parent companies. These are the most likely to be publicly traded.

In [9]:
companies_parents_df = companies_df[companies_df.id.isin(parent_ids)]
companies_parents_df.head()

Unnamed: 0  id             name    start_date  changed_company_id  parent   
0            0   1  Electronic Arts  3.913920e+08                 5.0     NaN  \
1            1   2          BioWare  7.915968e+08                 NaN     1.0   
4            4   8         2K Games  1.106611e+09                 NaN   139.0   
11          11  26      Square Enix  1.049155e+09                 NaN     NaN   
17          17  41       Riot Games  1.156982e+09                 NaN  3595.0   

    change_date  
0           NaN  
1           NaN  
4           NaN  
11          NaN  
17          NaN

In [10]:
companies_parents_df.head()

Unnamed: 0  id             name    start_date  changed_company_id  parent   
0            0   1  Electronic Arts  3.913920e+08                 5.0     NaN  \
1            1   2          BioWare  7.915968e+08                 NaN     1.0   
4            4   8         2K Games  1.106611e+09                 NaN   139.0   
11          11  26      Square Enix  1.049155e+09                 NaN     NaN   
17          17  41       Riot Games  1.156982e+09                 NaN  3595.0   

    change_date  
0           NaN  
1           NaN  
4           NaN  
11          NaN  
17          NaN

In [33]:
with open('./src/games_query.txt', 'r') as file:
    games_payload = file.read()

In [72]:
with open('./test.json', 'r') as file:
    json_response = json.load(file)

In [74]:
print(json_response[0])

{'id': 1, 'changed_company_id': 5, 'name': 'Electronic Arts', 'start_date': 391392000}


In [76]:
pd.json_normalize(
    json_response
)

id  changed_company_id                     name    start_date  parent   
0       1                 5.0          Electronic Arts  3.913920e+08     NaN  \
1       2                 NaN                  BioWare  7.915968e+08     1.0   
2       4              4940.0        Eidos Interactive  8.335008e+08    26.0   
3       5                 NaN  Interplay Entertainment  4.416768e+08     NaN   
4       8                 NaN                 2K Games  1.106611e+09   139.0   
..    ...                 ...                      ...           ...     ...   
495  1361                 NaN             Replay Games           NaN     NaN   
496  1363                 NaN                   Akella  7.572960e+08     NaN   
497  1364                 NaN         The Chinese Room  1.356912e+09   439.0   
498  1365                 NaN               Lucas Pope           NaN     NaN   
499  1368                 NaN          Impromptu Games           NaN     NaN   

      change_date  
0             NaN  
1             NaN  
2    1.257725e+09  
3             NaN  
4             NaN  
..            ...  
495           NaN  
496           NaN  
497           NaN  
498           NaN  
499           NaN  

[500 rows x 6 columns]

In [71]:
pd.read_json('./test.json')

id  changed_company_id                     name    start_date  parent   
0       1                 5.0          Electronic Arts  3.913920e+08     NaN  \
1       2                 NaN                  BioWare  7.915968e+08     1.0   
2       4              4940.0        Eidos Interactive  8.335008e+08    26.0   
3       5                 NaN  Interplay Entertainment  4.416768e+08     NaN   
4       8                 NaN                 2K Games  1.106611e+09   139.0   
..    ...                 ...                      ...           ...     ...   
495  1361                 NaN             Replay Games           NaN     NaN   
496  1363                 NaN                   Akella  7.572960e+08     NaN   
497  1364                 NaN         The Chinese Room  1.356912e+09   439.0   
498  1365                 NaN               Lucas Pope           NaN     NaN   
499  1368                 NaN          Impromptu Games           NaN     NaN   

      change_date  
0             NaN  
1             NaN  
2    1.257725e+09  
3             NaN  
4             NaN  
..            ...  
495           NaN  
496           NaN  
497           NaN  
498           NaN  
499           NaN  

[500 rows x 6 columns]

In [70]:
pd.json_normalize(json_response, sep='_')

id                                        age_ratings  aggregated_rating   
0  217  [{'id': 56705, 'rating': 8}, {'id': 56706, 'ra...          72.500000  \
1  432  [{'id': 20713, 'rating': 4}, {'id': 63768, 'ra...          88.000000   
2  438  [{'id': 20518, 'rating': 10}, {'id': 20519, 'r...          91.000000   
3  439  [{'id': 18201, 'rating': 5}, {'id': 20530, 'ra...          77.500000   
4  440  [{'id': 32487, 'rating': 4}, {'id': 63133, 'ra...          90.000000   
5  441  [{'id': 63134, 'rating': 10}, {'id': 117208, '...          89.833333   
6  442  [{'id': 20546, 'rating': 4}, {'id': 63132, 'ra...          83.333333   
7  506  [{'id': 28374, 'rating': 8}, {'id': 29431, 'ra...          75.100000   
8  513  [{'id': 30278, 'rating': 11}, {'id': 56328, 'r...          76.666667   
9  560  [{'id': 34071, 'rating': 11}, {'id': 34072, 'r...          74.000000   

   aggregated_rating_count  first_release_date  follows   
0                        4          1222300800        4  \
1                        8          1300147200       61   
2                        3          1095811200       85   
3                        2          1128643200        2   
4                        4          1163116800       60   
5                        7          1236038400       30   
6                        4          1266883200       17   
7                       13          1320105600       38   
8                        3          1341360000       18   
9                        6          1266278400       34   

                                          game_modes   
0               [{'id': 1, 'name': 'Single player'}]  \
1  [{'id': 1, 'name': 'Single player'}, {'id': 2,...   
2  [{'id': 1, 'name': 'Single player'}, {'id': 2,...   
3               [{'id': 1, 'name': 'Single player'}]   
4  [{'id': 1, 'name': 'Single player'}, {'id': 2,...   
5  [{'id': 1, 'name': 'Single player'}, {'id': 2,...   
6  [{'id': 1, 'name': 'Single player'}, {'id': 2,...   
7               [{'id': 1, 'name': 'Single player'}]   
8  [{'id': 2, 'name': 'Multiplayer'}, {'id': 3, '...   
9  [{'id': 1, 'name': 'Single player'}, {'id': 2,...   

                                              genres   
0  [{'id': 8, 'name': 'Platform'}, {'id': 12, 'na...  \
1  [{'id': 11, 'name': 'Real Time Strategy (RTS)'...   
2  [{'id': 11, 'name': 'Real Time Strategy (RTS)'...   
3  [{'id': 25, 'name': 'Hack and slash/Beat 'em u...   
4  [{'id': 11, 'name': 'Real Time Strategy (RTS)'...   
5  [{'id': 11, 'name': 'Real Time Strategy (RTS)'...   
6  [{'id': 11, 'name': 'Real Time Strategy (RTS)'...   
7  [{'id': 8, 'name': 'Platform'}, {'id': 31, 'na...   
8  [{'id': 12, 'name': 'Role-playing (RPG)'}, {'i...   
9                     [{'id': 5, 'name': 'Shooter'}]   

                                  involved_companies   
0  [{'id': 1162, 'company': {'id': 2, 'name': 'Bi...  \
1  [{'id': 227, 'company': {'id': 111, 'name': 'T...   
2  [{'id': 1111, 'company': {'id': 111, 'name': '...   
3  [{'id': 495, 'company': {'id': 111, 'name': 'T...   
4  [{'id': 811, 'company': {'id': 111, 'name': 'T...   
5  [{'id': 594, 'company': {'id': 111, 'name': 'T...   
6  [{'id': 940, 'company': {'id': 111, 'name': 'T...   
7  [{'id': 105789, 'company': {'id': 395, 'name':...   
8  [{'id': 104430, 'company': {'id': 112, 'name':...   
9  [{'id': 1357, 'company': {'id': 318, 'name': '...   

                                     name     rating  rating_count   
0  Sonic Chronicles: The Dark Brotherhood  63.382947            27  \
1                     Total War: Shogun 2  84.835269           153   
2                         Rome: Total War  82.281446           246   
3                  Spartan: Total Warrior  65.882986            14   
4                  Medieval II: Total War  86.496153           182   
5                       Empire: Total War  78.053679           121   
6                     Napoleon: Total War  76.770999            67   
7                       Sonic Generations  77.335185          

In [2]:
import requests

url = "https://opencritic-api.p.rapidapi.com/game/search"

querystring = {"criteria":"the withcer 3"}

headers = {
	"X-RapidAPI-Key": "20f8c23a10msh485f83f46b5522bp1e86dfjsnad9f5fcbd804",
	"X-RapidAPI-Host": "opencritic-api.p.rapidapi.com"
}

response = requests.get(url, headers=headers, params=querystring)

print(response.json())

[{'id': 463, 'name': 'The Witcher 3: Wild Hunt', 'dist': 0.615385}, {'id': 1557, 'name': 'The Witness', 'dist': 0.631579}, {'id': 13938, 'name': 'The Past Within', 'dist': 0.636364}, {'id': 188, 'name': 'The Evil Within', 'dist': 0.636364}, {'id': 11681, 'name': 'Within the Blade', 'dist': 0.652174}, {'id': 4502, 'name': 'The Evil Within 2', 'dist': 0.666667}, {'id': 6430, 'name': 'The Walker', 'dist': 0.684211}, {'id': 5147, 'name': 'Maria the Witch', 'dist': 0.695652}, {'id': 5701, 'name': 'The Wizards', 'dist': 0.7}, {'id': 13244, 'name': 'The Guild 3', 'dist': 0.7}]
